Sources/Confidence/EventStorage.swift

```diff
@@ -0,0 +1,143 @@
+import Foundation
+import os
+
+internal protocol EventStorage {
+    func startNewBatch()
+    func writeEvent(event: Event)
+    func batchReadyIds() -> [String]
+    func eventsFrom(id: String) -> [Event]
+    func remove(id: String)
+}
+
+internal class EventStorageImpl: EventStorage {
+    private static let DIRECTORY = "events"
+    private static let READYTOSENDEXTENSION = ".ready"
+    private let encoder = JSONEncoder()
+    private let decoder = JSONDecoder()
+    private let currentFolderURL: URL
+    private var currentFileURL: URL
+    private var currentBatch: [Event] = []
+
+    init() throws {
+        currentFolderURL = URL(fileURLWithPath: try EventStorageImpl.getFolderURL())
+        currentFileURL = currentFolderURL.appendingPathComponent("events-\(Date().currentTime)")
+    }
+
+    func startNewBatch() {
+        let latestWriteFile = latestWriteFile()
```

Comment:
`startNewBatch` should be able to just use the `currentFileUrl` and rename it to extension batchReady and then calling `resetCurrentFile` function. we need to call the same `resetCurrentFile` in init function which we are not right now doing that, making the current file to be lost (not batched) in the next session.